### PR TITLE
adding clarifications and examples for rbac/psp

### PR DESCRIPTION
**Pull request by puja108**
towards https://github.com/giantswarm/giantswarm/issues/2096

WIP

### CODE REVIEW

File: content/guides/securing-with-rbac-and-psp/index.md [hunk 1]
```diff
@@ -243,6 +243,77 @@ A user can be part of multiple groups and thus be assigned multiple roles, too.
...
+```yaml
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: development-admin
+  namespace: development
+subjects:
+  - kind: User
+    name: jane.w6wn8.k8s.ginger.eu-central-1.aws.gigantic.io
+    apiGroup: rbac.authorization.k8s.io
```
Comment (webwurst) [inline content/guides/securing-with-rbac-and-psp/index.md]: not needed/allowed here

File: content/guides/securing-with-rbac-and-psp/index.md [hunk 2]
```diff
@@ -243,6 +243,77 @@ A user can be part of multiple groups and thus be assigned multiple roles, too.
...
+  - kind: Group
+    name: dev-admin
+    apiGroup: rbac.authorization.k8s.io
```
Comment (webwurst) [inline content/guides/securing-with-rbac-and-psp/index.md]: not needed/allowed here

**Comment by webwurst** [comment]
just as an idea, you could show how to get the default `restricted` policy, so people can copy it to edit settings:
```
kubectl get psp restricted -o yaml --export
```

File: content/guides/securing-with-rbac-and-psp/index.md (hunk 1 above)
Comment (puja108) [reply content/guides/securing-with-rbac-and-psp/index.md]: not sure what you mean, official docs have that here: https://kubernetes.io/docs/admin/authorization/rbac/#role-binding-examples

File: content/guides/securing-with-rbac-and-psp/index.md (hunk 2 above)
Comment (puja108) [reply content/guides/securing-with-rbac-and-psp/index.md]: see above

**Comment by puja108** [comment]
RFR

File: content/guides/securing-with-rbac-and-psp/index.md (hunk 1 above)
Comment (webwurst) [reply content/guides/securing-with-rbac-and-psp/index.md]: well, super weird. `kubectl` complained and after removing it worked. need to check again..

Comment (webwurst) [reply content/guides/securing-with-rbac-and-psp/index.md]: ok, got it: it is fine for `RoleBinding` with`apiVersion: rbac.authorization.k8s.io/v1`, but not allowed on `v1beta1` :)

Comment (webwurst) [reply content/guides/securing-with-rbac-and-psp/index.md]: (but it is also fine to omit on `v1`, don't know what to make out of that..)

Comment (webwurst) [reply content/guides/securing-with-rbac-and-psp/index.md]: hm, in minikube with current kubernetes `v1.8.0` it also complains with `v1`..

Comment (webwurst) [reply content/guides/securing-with-rbac-and-psp/index.md]: ok! final findings: `kubectl explain rolebinding.subjects`
```
FIELDS:
   apiGroup	<string>
     APIGroup holds the API group of the referenced subject. Defaults to "" for
     ServiceAccount subjects. Defaults to "rbac.authorization.k8s.io" for User
     and Group subjects.
```
and i tried adding this to a `ServiceAccount` subject. so it is fine and also the default for `User` and `Group`. sorry for the confusion ,D

Comment (puja108) [reply content/guides/securing-with-rbac-and-psp/index.md]: ah, cool, thanks for going so deep on this, TIL <3

**Pull request merged**